internal/world/azure.go

```diff
@@ -0,0 +1,205 @@
+package world
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/Sirupsen/logrus"
+	"github.com/pkg/errors"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+)
+
+const (
+	AzureSubscriptionId string = "AZURE_SUBSCRIPTION_ID"
+	AzureClientId       string = "AZURE_CLIENT_ID"
+	AzureClientSecret   string = "AZURE_CLIENT_SECRET"
+	AzureKeyVaultUrl    string = "AZURE_KEY_VAULT_URL"
+	AzureApiVersion     string = "AZURE_API_VERSION"
+)
+
+type Oauth2Res struct {
+	AccessToken string `json:"access_token"`
+}
+
+type KeyVaultEntry struct {
+	Value string `json:"value"`
+}
+
+type SecretVersions struct {
+	Value []struct {
+		ID         string `json:"id"`
+		Attributes struct {
+			Enabled       bool   `json:"enabled"`
+			Created       int    `json:"created"`
+			Updated       int    `json:"updated"`
+			RecoveryLevel string `json:"recoveryLevel"`
+		} `json:"attributes"`
+	} `json:"value"`
+}
+
+type Azure struct {
+	logger         *logrus.Logger
+	Prefix         string
+	KeyMapping     map[string]string
+	keyVaultUrl    string
+	subscriptionId string
+	clientId       string
+	clientSecret   string
+	apiVersion     string
+	token          string
+}
+
+func (w *World) Azure() *Azure {
+	if w.azure != nil {
+		return w.azure
+	}
+	azureSubscriptionId := w.checkAzureEnv(AzureSubscriptionId)
+	azureClientId       := w.checkAzureEnv(AzureClientId)
+	azureClientSecret   := w.checkAzureEnv(AzureClientSecret)
+	azureKeyVaultUrl    := w.checkAzureEnv(AzureKeyVaultUrl)
+	azureApiVersion     := w.checkAzureEnv(AzureApiVersion)
+
+	if azureApiVersion == "" {
+		azureApiVersion = "7.0"
+	}
+
+	w.azure = &Azure{
+		logger:         w.logger,
+		KeyMapping:     make(map[string]string),
+		subscriptionId: azureSubscriptionId,
+		clientId:       azureClientId,
+		clientSecret:   azureClientSecret,
+		keyVaultUrl:    azureKeyVaultUrl,
+		apiVersion:     azureApiVersion,
+	}
+	return w.azure
+}
+
+func (w *World) checkAzureEnv(env string) string {
+	value := os.Getenv(env)
+	if w.logger != nil && value == "" {
+		w.logger.Warnf("%v not set.", env)
+	}
+	return value
+}
+
+func (a *Azure) Secret(path string) (string, error) {
+	prefixPath := fmt.Sprintf("%s%s", a.Prefix, path)
+	mapped, ok := a.KeyMapping[prefixPath]
+	if !ok {
+		mapped = path
+	}
+	// azure keyvault only allows alphanumeric chars and dashes, for easier use and backwards compatibility we
+	// replace "/" with "--" and use the double dashes as a separator
+	mapped = strings.Replace(mapped, "/", "--", -1)
```

Comment:
Given that we have to adjust existing templates, I think this can be removed. Rather we should error.

Reply:
I would leave it tbh, but i am not against removing it

Reply:
If I'd don't know about hashicorp vault, I'd say it's pretty arbitrary. Why slashes? Why the dash replacement? Why not replace other invalid characters, such as `|`?